### PR TITLE
Added forceRefit option in update.train - Fixes #445

### LINKED ISSUE #445: Correct Way of Using update.train?
@topepo Is this the correct way to continue training from an existing model? Seems like `update` only updates the model object to some most recent compatible version without actually calling fit using the new dataset? I think I am missing something important here. 

```
previousModel <- readRDS("....xxx.rds")  # load previously saved model object
previousModel$trainingData <- trainData # change training data to new data
newModel <- update(object = previousModel)
```

### PR DESCRIPTION
**Pull request by terrytangyuan**

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@zachmayer

File: pkg/caret/R/updates.R [hunk 1]
```diff
@@ -1,6 +1,6 @@
-update.train <- function(object, param = NULL, ...)
+update.train <- function(object, param = NULL, forceRefit = F, ...)
```
Comment (zachmayer) [inline pkg/caret/R/updates.R]: Let's spell it out all the way: `forceRefit = FALSE`

Comment (terrytangyuan) [inline pkg/caret/R/updates.R]: Good catch!

**Comment by zachmayer** [comment]
It looks like if `param = NULL` and `forceRefit=TRUE`, you'll need to extract the tuning parameters used by the final model for the retraining, e.g. `param <- object$bestTune`.

You could also force this behaivor in the current version of caret with `new_model <- update(old_model, param = old_model$bestTune`

**Comment by terrytangyuan** [comment]
Thanks for pointing out. But I don't see any success on my end though. After `update` is being called. It just finished instantly.

**Comment by terrytangyuan** [comment]
@zachmayer ping! updated code.

File: pkg/caret/R/updates.R [hunk 2]
```diff
@@ -1,5 +1,6 @@
-update.train <- function(object, param = NULL, ...)
+update.train <- function(object, param = NULL, forceRefit = FALSE, ...)
 {
+  if(forceRefit) param <- object$bestTune
```
Comment (zachmayer) [inline pkg/caret/R/updates.R]: I think this should be `if(forceRefit & is.null(param))` or even `if(is.null(param))`

Comment (zachmayer) [inline pkg/caret/R/updates.R]: Actually, let's go with `if(forceRefit & is.null(param))`

**Comment by zachmayer** [comment]
Needs a test though

**Comment by zachmayer** [comment]
LGTM.  needs a review from @topepo

**Comment by terrytangyuan** [comment]
@topepo Could you take a look?

**Pull request closed without merging**